src/InverseKinematics.cpp

```diff
@@ -0,0 +1,154 @@
+#include "InverseKinematics.h"
+
+
+InverseKinematics::InverseKinematics()
+:updated(false)
+,initialized(false)
+,alreadyOptimized(false)
+{
+    solverPointer = new InverseKinematicsIPOPT();
+    loader = IpoptApplicationFactory();
+    loader->Options()->SetStringValue("hessian_approximation", "limited-memory");
+}
+
+InverseKinematics::~InverseKinematics()
+{
+}
+
+bool InverseKinematics::prepareProblem(const std::string& filename, const std::vector< std::string >& consideredJoints, const iDynTree::Vector3& gains, const iDynTree::Vector3& desiredPosition, const iDynTree::Vector4& desiredQuaternion, const iDynTree::VectorDynSize& desiredJoints, const std::string& parentFrame, const std::string& endEffectorFrame)
+{
+    bool success;
+    success = solverPointer->loadFromFile(filename, consideredJoints);
+    if (success){
+        success = solverPointer->setFrames(parentFrame,endEffectorFrame);
+        if(success){
+            success = solverPointer->update(gains,desiredPosition, desiredQuaternion, desiredJoints);
+        }
+        else return false;
+    }
+    else return false;
+    
+   updated = true;
+   
+    return true;
+}
+
+bool InverseKinematics::setURDF(const std::string& filename, const std::vector< std::string >& consideredJoints)
```

Comment:
I thought that sometimes you just want to consider all joints. Actually those two functions calls the very same method of ``InverseKinematicsIPOPT`` class, just handling the availability of a list of considered joints.